### PR TITLE
Fat 1289 c17018

### PR DESCRIPTION
**Pull request by TetianaParanich**
***
Implement e-2-e automation of test case C17018: Check that field protection overrides work properly during data import
***
https://foliotest.testrail.io/index.php?/cases/view/17018
***

### CODE REVIEW

File: cypress/fixtures/oneMarcBib-BeforeOverride.mrc [hunk 1]
```diff
@@ -0,0 +1 @@
...
  a(DE-599)GBV880391235
00aAnglo-Saxon manuscripts in microfiche facsimile Volume 25 Corpus Christi College, Cambridge II, MSS 12, 144, 162, 178, 188, 198, 265, 285, 322, 326, 449 microform A. N. Doane (editor and director), Matthew T. Hussey (associate editor), Phillip Pulsiano (founding editor)
 1aTempe, ArizonabACMRS, Arizona Center for Medieval and Renaissance Studiesc2016
  a69 Mikrofiches 1 Begleitbuch (XII, 167 Seiten)
  atext
0 aMedieval and Renaissance Texts and Studies volume 497
0 avolume 497
  aThis is the ORIGINAL version of the non-repeatable 514 note
1 aLucas, Peter J
2 aArizona Center for Medieval and Renaissance Studies
```
Comment (IhorBohdan) [inline cypress/fixtures/oneMarcBib-BeforeOverride.mrc:1]: Does we need this files in fixtures?

Comment (TetianaParanich) [reply cypress/fixtures/oneMarcBib-BeforeOverride.mrc:1]: yes, I need upload 3 files in this case

File: cypress/support/fragments/data_import/job_profiles/newJobProfile.js [hunk 2]
```diff
@@ -95,6 +95,27 @@ export default {
...
+    ModalSelectActionProfile.selectActionProfile(matchProfileName, 'match');
+    cy.expect(Pane('New job profile').find(Accordion('Overview')).find(HTML(including(matchProfileName))).exists());
+    // link first action profile to match profile
+    cy.get('[id*="type-selector-dropdown-ROOT"]').eq(forMatchesOrder).click();
+    cy.do(actionsButton.click());
+    ModalSelectActionProfile.searchActionProfileByName(firstActionProfileName);
+    ModalSelectActionProfile.selectActionProfile(firstActionProfileName);
+    cy.expect(Pane('New job profile').find(Accordion('Overview')).find(HTML(including(firstActionProfileName))).exists());
+    // link second action profile to match profile
+    cy.get('[id*="type-selector-dropdown-ROOT"]').eq(forMatchesOrder).click();
```
Comment (IhorBohdan) [inline cypress/support/fragments/data_import/job_profiles/newJobProfile.js:112]: Maybe move to const cy.get

File: cypress/support/fragments/data_import/job_profiles/newJobProfile.js [hunk 3]
```diff
@@ -95,6 +95,27 @@ export default {
...
+    cy.do(actionsButton.click());
+    ModalSelectActionProfile.searchActionProfileByName(secondActionProfileName);
+    ModalSelectActionProfile.selectActionProfile(secondActionProfileName);
+    cy.expect(Pane('New job profile').find(Accordion('Overview')).find(HTML(including(secondActionProfileName))).exists());
+  },
+
   saveAndClose: () => {
     cy.do(Button('Save as profile & Close').click());
```
Comment (IhorBohdan) [inline cypress/support/fragments/data_import/job_profiles/newJobProfile.js:120]: Move button to const

File: cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js [hunk 4]
```diff
@@ -222,4 +228,35 @@ export default {
...
+      Select({ name:'profile.existingRecordType' }).choose(specialMappingProfile.typeValue),
+      Select({ name:'profile.mappingDetails.marcMappingOption' }).choose(fieldMappingsForMarc.update),
+    ]);
+  },
+
+  fillMappingProfileForInstance:(specialMappingProfile = defaultMappingProfile) => {
+    cy.do([
+      TextField({ name:'profile.name' }).fillIn(specialMappingProfile.name),
+      Select({ name:'profile.incomingRecordType' }).choose(incomingRecordType.marcBib),
+      Select({ name:'profile.existingRecordType' }).choose(specialMappingProfile.typeValue),
```
Comment (IhorBohdan) [inline cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js:245]: MOve select  to const

File: cypress/support/fragments/data_import/dataImport.js [hunk 5]
```diff
@@ -217,4 +218,15 @@ export default {
 
     cy.expect(deleteLogsButton.is({ disabled: true }));
   },
+
+  editMarcFile(editedFileName, finalFileName, stringToBeReplaced, replaceString) {
+    FileManager.readFile(`cypress/fixtures/${editedFileName}`)
+      .then((actualContent) => {
+        const content = actualContent.split('\n');
+        console.log(content);
```
Comment (penfriend) [inline cypress/support/fragments/data_import/dataImport.js:226]: do you need a console.log?

File: cypress/support/fragments/data_import/mapping_profiles/fieldMappingProfiles.js [hunk 6]
```diff
@@ -63,7 +64,7 @@ const deleteFieldMappingProfile = (profileName) => {
     });
 };
 
-const searchMappingProfileForDuplicate = (nameForSearch) => {
+const searchMappingProfile = (nameForSearch) => {
   cy.do(TextField({ id:'input-search-mapping-profiles-field' }).fillIn(nameForSearch));
   cy.expect(Button('Search').has({ disabled:false }));
```
Comment (penfriend) [inline cypress/support/fragments/data_import/mapping_profiles/fieldMappingProfiles.js:69]: move search button to const

File: cypress/support/fragments/data_import/mapping_profiles/mappingProfileDetails.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
+import { HTML } from '@interactors/html';
+import {
+  Accordion,
+  including,
+  MultiColumnListRow,
+  Button,
+  Pane,
+  Checkbox,
+  MultiColumnListCell,
+  EditableListRow
```
Comment (penfriend) [inline cypress/support/fragments/data_import/mapping_profiles/mappingProfileDetails.js:10]: please fix linter issue

File: cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js (hunk 4 above)
Comment (penfriend) [reply cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js:245]: Please move TextField({ name:'profile.name' }) to const too

Comment (TetianaParanich) [reply cypress/support/fragments/data_import/mapping_profiles/newMappingProfile.js:245]: these are different selects

File: cypress/support/fragments/data_import/dataImport.js (hunk 5 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/dataImport.js:226]: I deleted it

File: cypress/support/fragments/data_import/mapping_profiles/fieldMappingProfiles.js (hunk 6 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/mapping_profiles/fieldMappingProfiles.js:69]: Done

File: cypress/support/fragments/data_import/mapping_profiles/mappingProfileDetails.js (hunk 7 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/mapping_profiles/mappingProfileDetails.js:10]: Done

File: cypress/support/fragments/data_import/job_profiles/newJobProfile.js [hunk 8]
```diff
@@ -95,8 +95,29 @@ export default {
...
+    ModalSelectActionProfile.selectActionProfile(matchProfileName, 'match');
+    cy.expect(Pane('New job profile').find(Accordion('Overview')).find(HTML(including(matchProfileName))).exists());
+    // link first action profile to match profile
+    cy.get('[id*="type-selector-dropdown-ROOT"]').eq(forMatchesOrder).click();
+    cy.do(actionsButton.click());
+    ModalSelectActionProfile.searchActionProfileByName(firstActionProfileName);
+    ModalSelectActionProfile.selectActionProfile(firstActionProfileName);
+    cy.expect(Pane('New job profile').find(Accordion('Overview')).find(HTML(including(firstActionProfileName))).exists());
+    // link second action profile to match profile
+    cy.get('[id*="type-selector-dropdown-ROOT"]').eq(forMatchesOrder).click();
```
Comment (IhorBohdan) [inline cypress/support/fragments/data_import/job_profiles/newJobProfile.js:112]: cy.get to const

**Review by IhorBohdan — APPROVED** [review]

**Review by MaksimTerekhov — APPROVED** [review]

File: cypress/support/fragments/data_import/job_profiles/newJobProfile.js (hunk 8 above)
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/job_profiles/newJobProfile.js:112]: I can't do it

**Pull request merged**